src/solvers/quasi_Newton.jl

```diff
@@ -37,18 +37,20 @@ mutable struct QuasiNewtonState{
     T,
     D<:AbstractQuasiNewtonDirectionUpdate,
     SC<:StoppingCriterion,
-    S<:Stepsize,
+    S<:StepsizeStorage,
     RTR<:AbstractRetractionMethod,
     VT<:AbstractVectorTransportMethod,
 } <: AbstractGradientSolverState
     p::P
+    p_old::P
     X::T
     sk::T
     yk::T
     direction_update::D
     retraction_method::RTR
     stepsize::S
     stop::SC
+    new_grad::T
```

Comment:
can we stay consistent and use `p_old` and `X_old` and not mix old and new? I would be in favour of that.

Reply:
What name would you prefer? I used the name of variable in `step_solver!` that was used for the same purpose.

Reply:
I do not like the asymmetry between `X` (above) for the current gradient) and `_grad` here, not that we have `p_old` and `new_` (neither the switch in words nor the old/new usage). So in style I would prefer `X_old`.